impeller/entity/entity_pass.cc

```diff
@@ -585,9 +586,15 @@ EntityPass::EntityResult EntityPass::GetEntityForElement(
       auto texture = pass_context.GetTexture();
       // Render the backdrop texture before any of the pass elements.
       const auto& proc = subpass->backdrop_filter_proc_;
+
+      Matrix subpass_transform_basis = subpass->transform_.Basis();
       subpass_backdrop_filter_contents =
-          proc(FilterInput::Make(std::move(texture)),
-               subpass->transform_.Basis(), Entity::RenderingMode::kSubpass);
+          proc(FilterInput::Make(std::move(texture)), subpass_transform_basis,
+               // When the subpass has a translation that means the math with
+               // the snapshot has to be different.
+               subpass_transform_basis == subpass->transform_
+                   ? Entity::RenderingMode::kSubpassAppendSnapshotTransform
+                   : Entity::RenderingMode::kSubpassPrependSnapshotTransform);
```

Comment:
Could you add `Matrix::HasTranslation` for this? Similar to `Matrix::HasPerspective`, etc.

Reply:
Done.